src/main/capi/pending-c.cpp

```diff
@@ -82,6 +93,23 @@ duckdb_pending_state duckdb_pending_execute_task(duckdb_pending_result pending_r
 	}
 }
 
+duckdb_state duckdb_create_streaming_result(duckdb_pending_result pending_result, duckdb_result *out_result) {
```

Comment:
Is this extra method necessary? Can't we infer from having called `duckdb_pending_prepared_streaming` before that `duckdb_execute_pending` should return a streaming result?

Reply:
Yea we can, I just figured the api should be explicit, because they use the same type - could be confusing

But we can't really provide helpful errors in the C api anyways, so maybe its not a good idea